File: trunk/JAVA/PokerProtocol/src/bluffinmuffin/protocol/commands/lobby/career/CreateCareerTableResponse.java
package bluffinmuffin.protocol.commands.lobby.career;

import java.util.StringTokenizer;

import bluffinmuffin.protocol.commands.lobby.AbstractLobbyResponse;

public class CreateCareerTableResponse extends AbstractLobbyResponse<CreateCareerTableCommand>
{
    @Override
    protected String getCommandName()
    {
        return CreateCareerTableResponse.COMMAND_NAME;
    }
    
    public static String COMMAND_NAME = "lobbyCAREER_CREATE_TABLE_RESPONSE";
    private final int m_ResponsePort;
    
    public CreateCareerTableResponse(StringTokenizer argsToken)
    {
        super(new CreateCareerTableCommand(argsToken));
        m_ResponsePort = Integer.parseInt(argsToken.nextToken());
    }
    
    public CreateCareerTableResponse(CreateCareerTableCommand command, int responsePort)
    {
        super(command);
        m_ResponsePort = responsePort;
    }
    
    @Override
    public void encode(StringBuilder sb)
    {
        super.encode(sb);
        append(sb, m_ResponsePort);
    }
    
    public int getResponsePort()
    {
        return m_ResponsePort;
    }
    
}

File: trunk/C#/PokerProtocol/Commands/Lobby/Career/CreateCareerTableResponse.cs
using System;
using System.Collections.Generic;
using System.Text;
using EricUtility;
using EricUtility.Networking.Commands;
using Newtonsoft.Json.Linq;

namespace PokerProtocol.Commands.Lobby.Career
{
    public class CreateCareerTableResponse : AbstractLobbyResponse<CreateCareerTableCommand>
    {
        public static string COMMAND_NAME = "lobbyCAREER_CREATE_TABLE_RESPONSE";

        public int Port { get; set; }

        public CreateCareerTableResponse()
            : base()
        {
        }

        public CreateCareerTableResponse(CreateCareerTableCommand command, int port)
            : base(command)
        {
            Port = port;
        }
    }
}
